qiskit/transpiler/passes/basis/unroller.py

```diff
@@ -52,7 +52,7 @@ def run(self, dag):
             return dag
         # Walk through the DAG and expand each non-basis node
         for node in dag.op_nodes():
-            basic_insts = ['measure', 'reset', 'barrier', 'snapshot', 'delay']
+            basic_insts = ['measure', 'reset', 'barrier', 'delay']
```

Comment:
Is this because aer reports snapshot in it's basis? What about if someone uses the snapshot in terra? Will this cause unexpected results?

Reply:
Yeah. This change would make a backend without snapshots raise an exception if you try and run snapshot circuit when before it would just ignore the instruction.

I'm fine with leaving it in here as a special case though if you want to keep old behaviour

Reply:
Yeah I would switch it back just to minimize potential issues. I don't think it'll actually come up especially since the unroller isn't used by default anymore. This is really something we need to fix in the next iteration of the Backend object so we have a way for backends to describe the gates and instructions they support.

Reply:
Done